src/components/Tabs/Tabs.tsx

```diff
@@ -33,169 +36,141 @@ interface TabsState {
           ? NavButtonOptions.right
           : NavButtonOptions.both;
     }
 
-    return shouldShow;
-  };
+    return newNavButtons;
+  }
 
-  _onScroll = () => {
-    this._showNavButtonsIfNeeded();
-  };
+  _updateButtonsIfNeeded() {
+    const { navButtons } = this.state;
+    const newNavButtons = this._getNewNavButtons();
 
-  _scrollTabs(direction: number) {
-    const tabsElement = this._navRef.current;
-    const clientWidth = direction * tabsElement.clientWidth;
-    const nextScrollLeft = tabsElement.scrollLeft + clientWidth;
-    this._animateScroll(nextScrollLeft);
+    if (newNavButtons !== navButtons) {
+      this.setState({ navButtons: newNavButtons });
+    }
   }
 
-  _animateScroll(scrollLeft: number) {
-    animate('scrollLeft', this._navRef.current, scrollLeft);
-  }
+  _onResize = () => {
+    clearTimeout(this._resizeTimer);
 
-  _onNavClickLeft = () => {
-    this._scrollTabs(-1);
-  };
+    this._updateButtonsIfNeeded();
 
-  _onNavClickRight = () => {
-    this._scrollTabs(1);
+    this._resizeTimer = window.setTimeout(() => {
+      this.setState({ tabsKey: this._generateRandomKey() });
+    }, 100);
   };
 
-  _selectTab = (index: number) => {
-    const { activeTabIndex, onTabClick } = this.props;
+  _generateRandomKey() {
+    return `${Math.random()}`;
+  }
 
-    if (activeTabIndex !== index) {
-      onTabClick(index);
-    }
+  _tabsRefCallback = (el: ScrollableTabs) => {
+    this._tabsRef = el;
   };
 
   render() {
-    // const { mobile } = this.context;
     const { navButtons, tabsKey } = this.state;
-    const { items, activeTabIndex, skin, alignment, variant } = this.props;
-    const styleProps = { skin, alignment, variant, navButtons };
+    const { items, alignment, skin, variant, activeTabIndex } = this.props;
 
     return (
       <TPAComponentsConsumer>
         {({ mobile }) => (
-          <div {...style('root', { ...styleProps, mobile }, this.props)}>
+          <div
+            {...style('root', { skin, navButtons, mobile }, this.props)}
+            data-skin={skin}
+            data-navbuttons={navButtons}
+            data-mobile={mobile}
+          >
             <ReactResizeDetector handleWidth onResize={this._onResize} />
-            <TabsUI
-              key={tabsKey}
-              navRef={this._navRef}
-              selectedTabRef={this._selectedTabRef}
-              items={items}
-              onTabClick={this._selectTab}
-              activeTabIndex={activeTabIndex}
+            <ScrollableTabs
               alignment={alignment}
               variant={variant}
+              items={items}
+              className={style.navigation}
+              onClickItem={this._onClickItem}
               onScroll={this._onScroll}
-              onLeftNavClick={this._onNavClickLeft}
-              onRightNavClick={this._onNavClickRight}
+              activeTabIndex={activeTabIndex}
+              ref={this._tabsRefCallback}
+              key={tabsKey}
```

Comment:
If this key is only used for re-rendring `ScrollableTabs`, we might consider a different solution.